### PR TITLE
Don't flush after the request from publishAsync calls

### PR DESCRIPTION
**Pull request by scottf**

### CODE REVIEW

File: src/main/java/io/nats/client/impl/NatsConnection.java [hunk 1]
```diff
@@ -1205,15 +1205,15 @@ else if (future.isDone()) {
      */
     @Override
     public Message request(String subject, byte[] body, Duration timeout) throws InterruptedException {
-        return requestInternal(subject, null, body, timeout, cancelAction, true);
+        return requestInternal(subject, null, body, timeout, cancelAction, true, true);
```
Comment (scottf) [inline src/main/java/io/nats/client/impl/NatsConnection.java:1208]: All `public Message request(...)` are a user made request. The original behavior intended for the flush to happen after request hit the socket, hence the true.

File: src/main/java/io/nats/client/impl/NatsConnection.java [hunk 2]
```diff
@@ -1222,11 +1222,12 @@ public Message request(String subject, Headers headers, byte[] body, Duration ti
...
         validateNotNull(message, "Message");
-        return requestInternal(message.getSubject(), message.getHeaders(), message.getData(), timeout, cancelAction, false);
+        return requestInternal(message.getSubject(), message.getHeaders(), message.getData(), timeout, cancelAction, false, true);
     }
 
-    Message requestInternal(String subject, Headers headers, byte[] data, Duration timeout, CancelAction cancelAction, boolean validateSubjectAndReplyTo) throws InterruptedException {
-        CompletableFuture<Message> incoming = requestFutureInternal(subject, headers, data, timeout, cancelAction, validateSubjectAndReplyTo);
+    Message requestInternal(String subject, Headers headers, byte[] data, Duration timeout,
+                            CancelAction cancelAction, boolean validateSubjectAndReplyTo, boolean flushImmediatelyAfterPublish) throws InterruptedException {
+        CompletableFuture<Message> incoming = requestFutureInternal(subject, headers, data, timeout, cancelAction, validateSubjectAndReplyTo, flushImmediatelyAfterPublish);
```
Comment (scottf) [inline src/main/java/io/nats/client/impl/NatsConnection.java:1230]: Changed to accept the `flushImmediatelyAfterPublish` from callers

**Comment by johnlcox** [comment]
When will this be merged/released?

File: src/main/java/io/nats/client/Options.java [hunk 3]
```diff
@@ -1658,6 +1665,15 @@ public Builder useDispatcherWithExecutor() {
             return this;
         }
 
+        /**
+         * Instruct requests to turn off flush on requests.
+         * @return the Builder for chaining
+         */
+        public Builder dontForceFlushOnRequest() {
```
Comment (MauriceVanVeen) [inline src/main/java/io/nats/client/Options.java:1672]: personally not sure about using `dont`, maybe something like `disableForceFlushOnRequest` instead?

File: src/main/java/io/nats/client/impl/NatsConnection.java [hunk 4]
```diff
@@ -1239,31 +1240,31 @@ Message requestInternal(String subject, Headers headers, byte[] data, Duration t
      */
     @Override
     public CompletableFuture<Message> request(String subject, byte[] body) {
-        return requestFutureInternal(subject, null, body, null, cancelAction, true);
+        return requestFutureInternal(subject, null, body, null, cancelAction, true, true);
```
Comment (MauriceVanVeen) [inline src/main/java/io/nats/client/impl/NatsConnection.java:1243]: Should this and these below also be adapted to use `options.forceFlushOnRequest()`?

**Pull request merged**